src/raft.rs

```diff
@@ -572,6 +627,54 @@ impl<T: Storage> Raft<T> {
         self.raft_log.maybe_commit(mci, self.term)
     }
 
+    /// Commit that the Raft peer has applied up to the given index.
+    ///
+    /// Registers the new applied index to the Raft log.
+    ///
+    /// # Hooks
+    ///
+    /// * Post: Checks to see if it's time to finalize a Joint Consensus state.
+    pub fn commit_apply(&mut self, applied: u64) {
+        #[allow(deprecated)]
+        self.raft_log.applied_to(applied);
+
+        // Check to see if we need to finalize a Joint Consensus state now.
+        if let Some(index) = self.began_conf_change_at {
```

Comment:
What if the peer get down just after `Configuraion(old,new)` is just committed? Seems `began_conf_change_at` is not persisted with `Configuration(old,new)`.

Reply:
It's a great catch. You're right. If the Raft node goes down right now it will forget when it `began_conf_change_at`. We'll need to persist it.